lib/hmmlearn/base.py

```diff
@@ -641,6 +714,49 @@ def _initialize_sufficient_statistics(self):
                  'trans': np.zeros((self.n_components, self.n_components))}
         return stats
 
+    def _accumulate_sufficient_statistics_scaling(self, stats, X, frameprob,
```

Comment:
Instead of introducing a new function, can you not add a switch into the original _accumulate_sufficient_statistics, doing what's required depending on the value of `self.implementation`?

Reply:
I can do that if you would like.  The reason I went like this is that `frameprob` will be either probabilities, or log-probabilities, depending on which algorithm was used.  I imagined that being explicit with what is happening could be clearer.

If I undo this change, and bring back the single function, do you have a preference for the `frameprob` parameter, knowing that it can have two different interpretations based upon where it is called from ?

Reply:
Ah, I see, that the same problem as fwdlattice vs. alpha as noted below.  I guess we could call it `lattice` to go together with `fwdlattice` and `bwdlattice`?  Probably worth adding a comment somewhere noting that lattice/fwdlattice/bwdlattice correspond either to p/alpha/beta or to logp/logalpha/logbeta depending on implementation.

Reply:
OK, I have adjusted this interface - I resisted the urge to make all the names of the parameters in the derived classes be named the same.  We can save that for another day.

I'm open to further suggestions here.  Each of the fit routines could have called `accumulate_sufficient_statistics`, or they could return the needed data to have one call in `fit()`.  I chose the latter option, but could just as easily done the former.

Reply:
I think the current approach is fine for now.